File: app/src/main/java/info/nightscout/androidaps/plugins/PumpDanaR/comm/MsgHistoryNewDone.java
package info.nightscout.androidaps.plugins.PumpDanaR.comm;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import info.nightscout.androidaps.Config;
import info.nightscout.androidaps.Constants;

/**
 * Created by mike on 20.07.2016.
 */
public class MsgHistoryNewDone extends MessageBase {
    private static Logger log = LoggerFactory.getLogger(Constants.PUMPCOMM);
    public static boolean received = false;

    public MsgHistoryNewDone() {
        SetCommand(0x42F1);
        received = false;
        if (Config.logPumpComm)
            log.debug("New message");
    }

    @Override
    public void handleMessage(byte[] bytes) {
        received = true;
        if (Config.logPumpComm)
            log.debug("History new done received");
    }
}

File: pump/danar/src/main/java/info/nightscout/androidaps/danar/comm/MsgHistoryNewDone.kt
package info.nightscout.androidaps.danar.comm

import dagger.android.HasAndroidInjector
import info.nightscout.rx.logging.LTag


class MsgHistoryNewDone(
    injector: HasAndroidInjector
) : MessageBase(injector) {

    init {
        setCommand(0x42F1)
        danaPump.historyDoneReceived = false
        aapsLogger.debug(LTag.PUMPCOMM, "New message")
    }

    override fun handleMessage(bytes: ByteArray) {
        danaPump.historyDoneReceived = true
        aapsLogger.debug(LTag.PUMPCOMM, "History new done received")
    }
}